cmd/nginx-ingress/main.go

```diff
@@ -515,3 +525,19 @@ func getAndValidateSecret(kubeClient *kubernetes.Clientset, secretNsName string)
 	}
 	return secret, nil
 }
+
+const locationFmt = `/[^\s{};]*`
+const locationErrMsg = "must start with / and must not include any whitespace character, `{`, `}` or `;`"
+
+var locationRegexp = regexp.MustCompile("^" + locationFmt + "$")
+
+func validateLocation(location string) error {
```

Comment:
hmm I am wondering if, in the future, we should refactor all the validation methods in one single package to avoid code repetition? cc @pleshakov

Reply:
@Rulox 👍